explorer/explorermiddleware.go

```diff
@@ -35,6 +36,19 @@ func (exp *explorerUI) BlockHashPathOrIndexCtx(next http.Handler) http.Handler {
 				return
 			}
 		} else {
+                        //Handle blocks which are in future time
+                        maxHeight := exp.blockData.GetHeight()
+                        if height > int64(maxHeight) {
+                        maturityTime := int(float64(height - int64(maxHeight) ) * exp.ChainParams.TargetTimePerBlock.Minutes())
+                        days := maturityTime / 1440
+                        hours := (maturityTime % 1440  ) / 60
+                        minutes := maturityTime % 60
```

Comment:
In all the duration processing above, please make `maturityTime` (or `expectedTime`) be of type `time.Duration`.  Then you can do `minutes := expectedTime.Minutes()` to avoid most of the "magic numbers" above.  I usually prefer to use standard library functions when they are available.  Note `func (d Duration) Truncate(m Duration) Duration`